### PR TITLE
Reintegrate landing page into sphinx-generated documentation

### PR DESCRIPTION
**Pull request by romanlutz**
This also reenables the auto-push.

This includes all the changes from the fairlearn.github.io repository.

Will definitely not complete this PR before the end of the week, better only next week, to give people a chance to take a look.

### CODE REVIEW

**Comment by riedgar-ms** [comment]
I'm slightly confused by the description.... do you mean that it's incorporating the professionally designed homepage, which was previously only in the fairlearn.github.io repo?

File: docs/_templates/index.html [hunk 1]
```diff
@@ -0,0 +1,456 @@
+<!DOCTYPE html>
+<html lang="en">
+  	<head>
+	    <!-- Required meta tags -->
+	    <meta charset="utf-8">
+	    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+	    <title>FairLearn</title>
```
Comment (MiroDudik) [inline docs/_templates/index.html:7]: Fairlearn [this also needs changing on the current page]

File: notebooks/Grid Search for Binary Classification.ipynb [hunk 2]
```diff
@@ -6,7 +6,7 @@
    "source": [
     "# Grid Search for Binary Classification\n",
     "\n",
-    "This notebook goes through applying the `GridSearch` algorithm in FairLearn to a binary classification problem, where we also have a binary protected attribute. This algorithm comes from the paper [\"A Reductions Approach to Fair Classification\" (Agarwal et al. 2018)](https://arxiv.org/abs/1803.02453). The grid search is a simplified version of the full algorithm (appearing in section 3.4), which works best for binary classification and a binary protected attribute.\n",
+    "This notebook goes through applying the `GridSearch` algorithm in Fairlearn to a binary classification problem, where we also have a binary protected attribute. This algorithm comes from the paper [\"A Reductions Approach to Fair Classification\" (Agarwal et al. 2018)](https://arxiv.org/abs/1803.02453). The grid search is a simplified version of the full algorithm (appearing in section 3.4), which works best for binary classification and a binary protected attribute.\n",
```
Comment (riedgar-ms) [inline notebooks/Grid Search for Binary Classification.ipynb:9]: Not sure this quite belongs in this PR....

Comment (romanlutz) [reply notebooks/Grid Search for Binary Classification.ipynb:9]: I replaced all occurrences of FairLearn with Fairlearn. There was one outside of the documentation.

**Pull request merged**